models/double_entry_transactions/int_quickbooks__invoice_double_entry.sql

```diff
@@ -93,8 +127,63 @@ ar_accounts as (
     where account_type = '{{ var('quickbooks__accounts_receivable_reference', 'Accounts Receivable') }}'
         and is_active
         and not is_sub_account
+), 
+
+liability_accounts as (
+
+    select
+        account_id,
+        name,
+        source_relation
+    from accounts
+    where classification = 'Liability' 
+        and is_active
+),
+
+sales_tax_account as (
+
+    select
+        account_id,
+        source_relation
+    from accounts
+    where name = 'Sales Tax Payable' 
+        and is_active
+),
+
+global_tax_account as (
+
+    select
+        account_id,
+        source_relation
+    from accounts
+    where name = 'Global Tax Payable' 
+        and is_active 
 ),
 
+{% if var('using_tax_agency', True) %}
+tax_account_join as (
+
+    select 
+        tax_agencies.tax_agency_id,
+        tax_agencies.display_name,
+        coalesce(liability_accounts.account_id, sales_tax_account.account_id, global_tax_account.account_id) as account_id,
+        coalesce(liability_accounts.source_relation, sales_tax_account.source_relation, global_tax_account.source_relation) as source_relation
+
+    from tax_agencies
+    
+    left join liability_accounts
+        on {{ dbt.concat([
+            "tax_agencies.display_name", 
```

Comment:
Let's include a DECISIONLOG.md entry detailing why we've decided to take this approach.

Reply:
It may be worthwhile to document in detail how this mapping works for greater transparency and understanding.